### PR TITLE
Resuming NS runs

### PR DESCRIPTION
**Pull request by jacoterh**
This PR implements the option to resume from previous unfinished runs. 
- The docs have been updated accordingly to document the new settings: https://github.com/LHCfitNikhef/smefit_release/blob/intermediate_output/docs/source/fitting_code/running.md
- Only global fits can be resumed. No need for individual fits as these are fast anyway.

### CODE REVIEW

File: src/smefit/optimize/ultranest.py [hunk 1]
```diff
@@ -128,39 +127,61 @@ def from_dict(cls, config):
...
+        if "frac_remain" not in ns_settings["sampling_settings"]:
+            frac_remain_def = 0.01
             _logger.warning(
-                f"Remaining fraction (frac_remain) not set in the input card. Using default: {frac_remain}"
+                f"Remaining fraction (frac_remain) not set in the input card. Using default: {frac_remain_def}"
             )
+            ns_settings["sampling_settings"]["frac_remain"] = frac_remain_def
 
-        store_raw = config.get("store_raw", False)
+        #store_raw = ns_settings.get("store_raw", False)
```
Comment (giacomomagni) [inline src/smefit/optimize/ultranest.py:184]: If I recall correctly storing the intermediate results was quite heavy in memeory.
Is there still the possibility of not storing the partial results?

`store_raw` is used also below to determine the `log_dir`

File: src/smefit/optimize/ultranest.py [hunk 2]
```diff
@@ -71,12 +71,7 @@ def __init__(
         single_parameter_fits,
         pairwise_fits,
         use_multiplicative_prescription,
-        live_points=500,
-        lepsilon=0.001,
-        target_evidence_unc=0.5,
-        target_post_unc=0.5,
-        frac_remain=0.01,
-        store_raw=False,
+        ns_settings
```
Comment (giacomomagni) [inline src/smefit/optimize/ultranest.py:74]: A bit more picky comment, the whole point of having the `from_dict` method was for to group variables in the runcards, but leaving them spelled out in the class constructor to make more clear, which are the relevant one. 
I'd prefer to keep the default values here.

**Comment by LucaMantani** [comment]
@giacomomagni @jacoterh I would retrieve this PR. Aside for resuming runs, which is very helpful for long ones, I particularly like that we have all the ultranest options available automatically without need to spelling them out. This way, we can take full advantage of the numerous options provided without having to modify the code.

**Comment by LucaMantani** [comment]
Let's close this for now and let's see in the future if needed.

**Pull request closed without merging**